impl/src/main/java/org/ehcache/spi/serialization/DefaultSerializationProvider.java

```diff
@@ -158,12 +158,22 @@ private AbstractProvider(Map<Class<?>, Class<? extends Serializer<?>>> serialize
     @Override
     public <T> Serializer<T> createKeySerializer(Class<T> clazz, ClassLoader classLoader, ServiceConfiguration<?>... configs) throws UnsupportedTypeException {
       DefaultSerializerConfiguration<T> conf = find(SerializerConfiguration.Type.KEY, configs);
+      if(conf != null) {
```

Comment:
This had to be done since serialisation provider classes does not extend `ClassInstanceProvider` anymore. So this workaround was required. If I manage to refactor `DefaultSerializationProvider` to extend `ClassInstanceProvider` again, this special handling won't be required.